With UK layouts, the grave key should operate as a dead key.

diff -uNr xkeyboard-config-1.7.orig/symbols/gb xkeyboard-config-1.7/symbols/gb
--- xkeyboard-config-1.7.orig/symbols/gb	2009-08-15 23:31:43.000000000 +0900
+++ xkeyboard-config-1.7/symbols/gb	2011-05-19 12:56:47.000000000 +0900
@@ -151,7 +151,7 @@
     key <AE03> { [	    3,	sterling,  threesuperior, NoSymbol	] };
     key <BKSL> { [ numbersign,	asciitilde	] };
     key <LSGT> { [  backslash,	bar		] };
-    key <TLDE> { [	grave,	notsign,   bar,           bar		] };
+    key <TLDE> { [ dead_grave,	notsign,   bar,           bar		] };
 };
 
 // Dvorak letter positions, but punctuation all in the normal UK positions.
